rskj-core/src/main/java/co/rsk/net/handler/txvalidator/TxValidatorMaximumGasPriceValidator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2022 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.net.handler.txvalidator;
+
+import co.rsk.core.Coin;
+import co.rsk.net.TransactionValidationResult;
+import co.rsk.validators.TxGasPriceCap;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.ethereum.core.AccountState;
+import org.ethereum.core.Transaction;
+
+import javax.annotation.Nullable;
+import java.math.BigInteger;
+
+/**
+ * Validates that transaction's gas price is below the maximum allowed
+ */
+public class TxValidatorMaximumGasPriceValidator implements TxValidatorStep {
+
+    private final ActivationConfig activationConfig;
+
+    public TxValidatorMaximumGasPriceValidator(ActivationConfig activationConfig) {
+        this.activationConfig = activationConfig;
+    }
+
+    @Override
+    public TransactionValidationResult validate(Transaction tx, @Nullable AccountState state, BigInteger gasLimit, Coin minimumGasPrice, long bestBlockNumber, boolean isFreeTx) {
+        boolean isRskip252Enabled = activationConfig.isActive(ConsensusRule.RSKIP252, bestBlockNumber);
```

Comment:
I wonder if, in the future, we better cache the activation states for all consensus rules before calling validate( tx...) because isActive() will be called for each transaction (generating a activationHeights.get(consensusRule)). This will be costly if a block has 1000 transactions.
Instead, we could call validate(tx..., activeRSKIPs) where activeRSKIPs is a struct containing booleans for each relevant RSKIP.